[Houten Teo] iP

houtenteo opened the pull request:
## Duke

> This stuff is good man - me, 2021

### Can't remember what you need to do? 
### Duke can help you with that :)

#### Duke is:
  * Text-based
  * Easy to learn
  * Easy to use 
  * Fast
  * All around *goodness*
  * For a cheap price of _**$0**_

#### To get started, simply follow these x simple steps:
  1. [Download it](https://github.com/houtenteo/ip/releases/tag/v1)
  2. Navigate to the folder which you downloaded it to
  3. Open your terminal
  4. Enter the command 'java -jar -ip.jar'
  5. And have fun :relaxed: 
  
#### Features:
  - [x] Add new tasks
  - [x] Delete tasks
  - [x] Mark tasks as completed
  - [x] Searching for specific tasks 
  - [ ] Cooking your lunch
  - [ ] Waking you up in the morning 

#### Here's a `sneak peak` under the covers :wink: :
```java
public static void main(String[] args) {
        new Duke().start();
}
```

Commit:
Add Gradle support

houtenteo committed:
first commit

houtenteo committed:
added greet, echo exit

houtenteo committed:
Created 'List' class

houtenteo committed:
Added some functionalities

houtenteo committed:
Added the list feature

houtenteo committed:
Level-2

houtenteo committed:
Redundant class

Relalised there was no point for this clas

houtenteo committed:
Level-2

Removed the redundant list class

houtenteo committed:
Added the 'Task' class

houtenteo committed:
updated markComplete

houtenteo committed:
added getStatusIcon method

houtenteo committed:
Level 3

houtenteo committed:
Improved println

houtenteo committed:
Level-3

Tweaked a little bit to allow easier transition into level-4

houtenteo committed:
Created a "Todo" class

houtenteo committed:
Created a "MyList" class

to abstract out the print statements

houtenteo committed:
Changed the print statements a little

houtenteo committed:
Added 'todo' command

houtenteo committed:
Added 'Deadline' class

houtenteo committed:
Added 'toString()' method

houtenteo committed:
updated 'addTask' and 'listall' to incorporate toString method

houtenteo committed:
Update Duke.java

added deadline command

houtenteo committed:
Added wrongformatexception

houtenteo committed:
Added "Event" class

houtenteo committed:
Added overidden "getStatusIcon()" and "markComplete()"

houtenteo committed:
added event command

houtenteo committed:
added indicator for number of items in list after adding a task

houtenteo committed:
improved error handling

houtenteo committed:
Level-4

houtenteo committed:
Added my own inputs

houtenteo committed:
Expected text based on my input

houtenteo committed:
removed redundant lines in the bot UI

houtenteo committed:
testing new inputs

houtenteo committed:
added error handling for 'todo'

houtenteo committed:
imrpoved error handling

houtenteo committed:
testing error handling messages

houtenteo committed:
Improved message for trying to complete a completed task

houtenteo committed:
override the getStatusIcon() from the Task class

houtenteo committed:
Added delete functionality

houtenteo committed:
Added documentation

houtenteo committed:
Add new branch feature1

houtenteo committed:
test master branch

houtenteo committed:
Merge branch 'master' into feature1

houtenteo committed:
add another commit to feature1 branch

houtenteo committed:
add another commit in master branch

houtenteo committed:
Merge branch 'feature1'

houtenteo committed:
create add-countries file in add-countries branch

houtenteo committed:
merge add-countries into master

houtenteo committed:
Merge branch 'add-countries'

houtenteo committed:
create merge conflict with master

houtenteo committed:
create merge conflict with add-countries

houtenteo committed:
resolve merge conflict

houtenteo committed:
Add file to store list data

houtenteo committed:
Add file to store data in list

houtenteo committed:
Add getTaskFromString Method

This method converts the task in string format in the list to the appropriate task object

houtenteo committed:
Add constructor

Constructor transfers the tasks in the Data,txt file into the list if any

houtenteo committed:
Empty data file

houtenteo committed:
Add LocalDate variable

houtenteo committed:
Add static variable format

Determines the I/O format of the dates

houtenteo committed:
Add getFormat and SetFormat

Allow users to change the date format specifiers

houtenteo committed:
Add 'format' command

'format' command allows users to view the current date format

houtenteo committed:
Merge branch 'branch-Level-8'

houtenteo committed:
Remove format initializer

houtenteo committed:
Add default format

This will allow users to override and save their preferred format using setFormat

houtenteo committed:
Improve writeToFile() method

houtenteo committed:
Improve getTaskFromString() method

houtenteo committed:
Improve logic

now additionally stores the date format at the time of creating the task into the data.txt file

houtenteo committed:
change writeToFile call

houtenteo committed:
Improve writeToFile logic

houtenteo committed:
Add parser class

Abstracts out the logic of reading the user input

houtenteo committed:
Add Ui class

Abstracts out the logic of the Ui

houtenteo committed:
Add storage class

Deals with the uploading and downloading of the data to and from the Data.txt file

houtenteo committed:
Improve to accommodate the new classes

houtenteo committed:
Improve to accommodate the new classes

houtenteo committed:
Rearrange files into packages

all files moved into the 'duke' package.
within the duke package, wrongCommandFormatException was placed under the exception package and task, todo, deadline, event were placed under the tasktype package

houtenteo committed:
reorganise input statements

houtenteo committed:
Add initialisation of default date

houtenteo committed:
Add DeadlineTest class

houtenteo committed:
Add StorageTest class

houtenteo committed:
Change location of Data.txt file

houtenteo committed:
Add documentation

houtenteo committed:
Add documentation

houtenteo committed:
Change to interface

houtenteo committed:
Update to fit task interface

houtenteo committed:
Abstract out user input into parser class

houtenteo committed:
Abstract out system messages

houtenteo committed:
Abstract out Ui messages

houtenteo committed:
Add method to print more messages

Mainly for the MyList, deadline, event and todo classes

houtenteo committed:
Add documentation

houtenteo committed:
Update test cases

Changed the constructor of the Storage class. Had to include the file path in the constructor.

houtenteo committed:
Align with coding standard

houtenteo committed:
Add find() method

houtenteo committed:
Add 'find' command

houtenteo committed:
Add new messages for 'find' command

houtenteo committed:
Improve format

houtenteo committed:
Add documentation

houtenteo committed:
Merge branch 'add-gradle-support'

houtenteo committed:
Add gradle

houtenteo committed:
remove redundant files

houtenteo committed:
rename package to 'testing'

For easier differentiation from the gradle ip.test directory

houtenteo committed:
Add dependencies for javafx

houtenteo committed:
Add launcher class

Handles the launch of the app

houtenteo committed:
Extend Application class

houtenteo committed:
Create Gui file to format the GUI

houtenteo committed:
Create controller class

Handles the different action events

houtenteo committed:
Update listAll method to return a string

To enable the list to be displayed on the GUI

houtenteo committed:
Update the listAllMessage method to return a string

Align with the change to the change to the listAll method in MyList class

houtenteo committed:
Add gui fields and methods

enter method tests if the enter key does what i want it to.
listAll method checks to see if the controller class can communicate with the MyList class properly

houtenteo committed:
Add listAll button functionality

houtenteo committed:
Delete Gui.fxml

houtenteo committed:
Add fxml files

houtenteo committed:
Change loaded fmxl file

houtenteo committed:
Add getUserDialog and getDukeResponse method

houtenteo committed:
Delete DialogBox.fxml

houtenteo committed:
Update methods to return strings instead of printing to the screen

This is so that the responses from duke can be displayed on the GUI instead of being printed to the terminal

houtenteo committed:
Update mainWindow Gui layout

houtenteo committed:
Remove println statement

houtenteo committed:
Change gui layout

houtenteo committed:
Add DialogBox class

Abstracted out the dialogboxes into its own class

darrenhoon commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+     * Method to get the current format setting.
+     * @return the DateTimeFormatter
+     */
+    public static String getFormat() {
```
great to see that you format your response to enhance readability of your information for your users!

darrenhoon commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,142 @@
+ * @author Houten Teo
+ * @version CS2103T week 3
+ */
+public class Parser {
```
great to see that you placed the middleware logic into a Parser file. Exceptional quality of work that shows foresight into how you considered the project structure might become and this ensures that your project is modular, where people can easily contribute to your project using existing code.

darrenhoon commented on src/main/java/duke/exception/WrongCommandFormatException.java:
```diff
@@ -0,0 +1,15 @@
+package duke.exception;
+
+public class WrongCommandFormatException extends Exception{
```
Perhaps there can be a header comment here to inform users on what the class does. Similar comment for the other classes, if any, lacking a comment header

houtenteo replied:
Oops missed that out. Thanks for spotting it.

darrenhoon commented on src/main/java/duke/tasktype/Todo.java:
```diff
@@ -0,0 +1,113 @@
+ * @author Houten Teo
+ * @version CS2103T week 3
+ */
+public class Todo implements Task {
```
Great to see that you are using inheritance and polymorphism! That should help reduce duplicate code in your codebase and make your project a lot cleaner!

darrenhoon commented on src/test/java/test/DeadlineTest.java:
```diff
@@ -0,0 +1,59 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class DeadlineTest {
```
Excellent work incorporating testing into your project to ensure your project is robust!

darrenhoon reviewed (commented):
Overall, the quality of your code is high and it shows that you thought through a lot on how your codebase should be.

Onboarding for your project should be rather easy thanks to the extensive work you put in to document your codebase in a clear and insightful manner.

Looks **Very** Good To Me!

houtenteo committed:
Improve duke replies

houtenteo committed:
Remove redundant commented out code

houtenteo committed:
Add isUser parameter

To allow the customisation of the dialog box depending on whether its a user input or duke reply

houtenteo committed:
Update DialogBox constructor call

houtenteo committed:
Change javaFx version to 11 to align with jdk11

houtenteo committed:
Made the stage unresizable

houtenteo committed:
remove redundant print statements

houtenteo committed:
Move test files into ip.test

houtenteo committed:
Remove conflicting merge lines

houtenteo committed:
Merge branch 'branch-Level-10-GUI'

houtenteo committed:
Remove redundant commented out code

houtenteo committed:
Merge branch 'branch-Level-10'

houtenteo committed:
Import checkstyle files from AB3

houtenteo committed:
Fix checkstyle issues

houtenteo committed:
Fix CheckStyle errors

houtenteo committed:
Update methods

The Ui class methods were originally in-charge of printing to the screen duke's responses.
However, since the introduction of the GUI version, the Ui class methods have all been converted to returning Strings instead of printing to the screen.
Now, the print functionality is added back into the methods in order to allow the CLI version of duke to continue functioning.

houtenteo committed:
Add method to launch the CLI version of duke

houtenteo committed:
Add assert

houtenteo committed:
Improve method names

houtenteo committed:
Extract method from getDukeResponse

The original getDukeResponse method was long and hard to read.
The generation of the response for each specific command has been abstracted out into their respective methods instead of having the getDukeResponse method cover for everything.
This should greatly improve readability.

houtenteo committed:
Remove duplicate code

Originally, the readInput method was meant for the CLI version of Duke, while the getDukeResponse method was meant for the GUI version.
However, since both methods are very similar, getDukeResponse has now been updated to achieve the same outputs for the CLI version as well. Hence there is no need for readInput anymore.

houtenteo committed:
Improve startCli method

houtenteo committed:
Add distinction in CLI and GUI duke launch methods

houtenteo committed:
Add duke and user icons

houtenteo committed:
Update controller.java

Made use of the user and duke icons

houtenteo committed:
Remove unused fields and import statements

houtenteo committed:
Add assert

The String gotten from the data file should contain at least 6 characters representing '[ ][ ]'.
If not, it suggests that there might be an bug with the writeToFile method.

houtenteo committed:
Merge pull request #1 from houtenteo/branch-A-Assertions

Branch a assertions

houtenteo committed:
Merge branch 'master' into branch-A-CodeQuality

houtenteo committed:
Merge pull request #2 from houtenteo/branch-A-CodeQuality

Branch a code quality

houtenteo committed:
Add getTaskType method

In addition, also added a final field for the respective task subclasses.

houtenteo committed:
Add findDeadlineWithin method

This method helps with generating duke's response to the 'remind' command that is going to be implemented.

houtenteo committed:
Add methods for 'remind' command

These methods help to generate duke's responses to the 'remind' command.

houtenteo committed:
Add remind command and organised code stucture

Originally, the helper functions took in a string. However, this led to some problems when there was no nextLine in the scanner for this string to be generated.
Since it is more convenient for the helper methods to handle this exception, I have updated the scanner to become a class variable to allow the helper methods to access it.

houtenteo committed:
Fix checkstyle errors

houtenteo committed:
Improve markComplete messages

houtenteo committed:
Add isDone method

houtenteo committed:
Improve findDeadlineWithin method

Originally this method still displayed deadlines that have been completed when the user inputs the 'remind' command.
Now if, the deadline has already been completed, it wont show up when the user inputs the 'remind' command

houtenteo committed:
Improve withinDeadlineMessage method

Now the method would inform the user if certain deadlines are overdue and by how much.

houtenteo committed:
Add initialize method in Controller

Originally the user would be greeted by a blank screen but this method allows the welcome message to be displayed as soon as the app is started.

houtenteo committed:
Improve code readability

houtenteo committed:
Remove unused import statements

houtenteo committed:
Add 'clear' command that clears all items in the list

houtenteo committed:
Add title to GUI window

houtenteo committed:
Improve Gui

Added borders between the dialog boxes of duke and the user

houtenteo committed:
Update README.md

houtenteo committed:
Add representative screenshot

houtenteo committed:
Merge branch 'master' of https://github.com/houtenteo/ip

houtenteo committed:
Update javadoc to comply with coding standard

houtenteo committed:
Update README.md

houtenteo committed:
Update README.md

houtenteo committed:
Add help command

houtenteo committed:
Update README.md

houtenteo committed:
Delete Data.txt

houtenteo committed:
Merge branch 'master' of https://github.com/houtenteo/ip

houtenteo committed:
Update README.md

houtenteo committed:
Update README.md

houtenteo committed:
Update README.md

houtenteo committed:
Move to resources package

houtenteo committed:
Change to relative path

houtenteo committed:
Add javadoc

houtenteo committed:
Improve code quality

houtenteo committed:
Improve code quality

houtenteo committed:
Add javadoc

houtenteo committed:
Improve code quality

houtenteo committed:
Improve method names to fit coding standard

houtenteo committed:
Move to resources folder

houtenteo committed:
Move to resources folder

houtenteo committed:
Change how file path is obtained

Originally used the URL method, but realised it would not work in JAR format. Hence it was changed to getResources and getResourcesAsStream

houtenteo committed:
Add javadoc

houtenteo committed:
Remove commented out code

houtenteo committed:
Update build.gradle
